test: improve branch and line coverage for useDebounce hook

iamnikhilranjan opened the pull request:
**What kind of change does this PR introduce?**

Test improvement / coverage enhancement

**Issue Number:**

Fixes #3615

**Snapshots/Videos:**

<img width="807" height="73" alt="image" src="https://github.com/user-attachments/assets/f83c135a-8343-41f1-98a0-95338c65a517" />

**If relevant, did you update the documentation?**

No documentation updates required.

**Summary**

This PR improves test coverage for the `useDebounce` hook located in 
`src/components/OrgListCard/useDebounce.tsx`.

Previously, the branch coverage was incomplete because some internal 
conditional paths inside the hook (especially the cancel logic and 
the null-timeout branch) were not exercised by the test suite.

This PR introduces additional tests to fully cover:

• All branches in the `cancel` callback
• Delay updates and closure regeneration in the hook
• Debounce timeout cleanup behavior

With these additions, the file now achieves 100% coverage across 
statements, branches, functions, and lines.

**Does this PR introduce a breaking change?**

No — this PR modifies tests only.

## Checklist

### CodeRabbit AI Review
- [ ] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [ ] I have implemented or provided justification for each non-critical suggestion
- [ ] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [x] I have written tests for all new changes/features
- [x] I have verified that test coverage meets or exceeds 95%
- [x] I have run the test suite locally and all tests pass


**Other information**

<!--Add extra information about this PR here-->

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Tests**
  * Added three tests for debounce behavior: verifies updating the delay changes subsequent debounced calls, confirms cancel prevents pending callbacks, and ensures cancel with no pending timeout has no effect while debounce still works. Refactored test flow and removed an outdated test comment.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

palisadoes commented:
1. This file already is at 100% code coverage.
2. What was the rationale for the updates?

iamnikhilranjan commented:
> 1. This file already is at 100% code coverage.
> 2. What was the rationale for the updates?


I updated the tests because , a couple of branch paths in the hook weren’t being exercised.

Specifically:

* The `cancel()` branch where `timeoutRef.current` is `null`:

```ts
if (timeoutRef.current !== null) {
  clearTimeout(timeoutRef.current);
  timeoutRef.current = null;
}
```

* And the case where the hook re-creates the debounced callback after the `delay` changes:

```ts
const debouncedCallback = useCallback(
  (...args) => { ... },
  [callback, delay]
);
```

The new tests just make sure those paths run.
I also cleaned up some test names based on the review feedback.
